src/k8s-engine/images/imageFactory.ts

```diff
@@ -1,26 +1,29 @@
 import { ImageProcessor } from '@/k8s-engine/images/imageProcessor';
 import NerdctlImageProcessor from '@/k8s-engine/images/nerdctlImageProcessor';
+import MobyImageProcessor from '@/k8s-engine/images/mobyImageProcessor';
+import { ContainerEngine } from '@/config/settings';
 import * as K8s from '@/k8s-engine/k8s';
 
-/**
- * An or-barred enum of valid string values for the names of supported image processors
- */
-export type ImageProcessorName = 'nerdctl'; // | 'kim' has been dropped
+const cachedImageProcessors: Partial<Record<ContainerEngine, ImageProcessor|null>> = { };
```

Comment:
Interesting that we still need a type-assertion in the return stmt.

Reply:
Because it's a `Partial<T>`, so `cachedImageProcessors[engineName]` could potentially be `undefined` (key is missing).  I assume if we did `cachedImageProcessors[engineName]!` it'd be fine, but that triggers lint.  We can rewrite the code to make it go away, but it's neither shorter nor more readable.